bin/upload_sarif_to_gh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -eu
+
+sarif_file=$1
+owner=$2
+repo=$3
+
+if [ -z "$sarif_file" ] || [ -z "$owner" ] || [ -z "$repo" ]; then
+  echo "No argument provided. Usage: ./upload_sarif_to_gh.sh <path to .sarif report> <gh owner> <gh repo>"
+  exit 1
+fi
+
+gzip -c "$sarif_file" | base64 >sarif_base64.tmp
```

Comment:
Could be worth adding more error handling. For example:
 - check that the `$sarif_file` exists / is a regular file / is readable before calling `gzip`
 - check early that `$GITHUB_TOKEN` is defined and not empty, to exit early instead of having the error in those cases only happen during `curl` at the end of the script.

You might also want to `set -o pipefail` (or update your shebang to `#!/bin/bash -euo pipefail`) so that errors in a `|` pipe (like here where `gzip` output is piped thru `base64`) are not swallowed by the pipe and ignored. To be honest, I'm actually surprised that `shellcheck` (that is run on CI as part of the call to `make lint` in this repo's pipeline) didn't recommend this when linting that script 🤔